lib/Middleware/PreviewMiddleware.php

```diff
@@ -0,0 +1,91 @@
+<?php
+
+namespace OCA\PreviewGenerator\Middleware;
+
+use OC\Core\Controller\PreviewController;
+use OC\OCS\Exception;
+use OCP\IDBConnection;
+use OCP\AppFramework\Http;
+use OCP\AppFramework\Middleware;
+use OCP\Files\NotFoundException;
+use OCP\AppFramework\Http\DataResponse;
+
+class PreviewMiddleware extends Middleware {
+
+    /** @var IDBConnection */
+    private $connection;
+
+    public function __construct(IDBConnection $connection) {
+        $this->connection = $connection;
+    }
+
+    public function afterException($controller, $methodName, \Exception $exception) {
+        if ($exception instanceof NotFoundException) {
+            return new DataResponse([], Http::STATUS_NOT_FOUND);

```

Comment:
Better use existing NotFoundResponse in OCP\AppFramework\Http

Reply:
https://github.com/AlexSat/previewgenerator/pull/2